specs/factorial_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'minitest/autorun'
+require 'minitest/reporters'
+require_relative '../lib/factorial'
+
+describe "factorial" do
+  describe "basic tests" do
+    describe "test 1" do
+      it "factorial(5) = 120" do
+        factorial(5).must_equal 120
+      end
+    end
+
+    describe "test 2" do
+      it "factorial(7) = 5040" do
+        factorial(7).must_equal 5040
+      end
+    end
+  end
+
+  # if the parameter is an object, check for nil
+  describe "nil object input test" do
+    it "nil object is not an integer" do
```

Comment:
I think we could move this nil input test case into the edge cases section.

Reply:
Done!